Mathlib/Algebra/MvPolynomial/Equiv.lean

```diff
@@ -455,6 +455,41 @@ lemma totalDegree_coeff_optionEquivLeft_le
 
 end
 
+section
+
+variable (α : Type*)
+
+/-- `R[Xᵢ, Y]` is an `R[Xᵢ]`-algebra. -/
+noncomputable def algebraOption : Algebra (MvPolynomial α R) (MvPolynomial (Option α) R) :=
+  (MvPolynomial.rename some).toAlgebra
+
+attribute [local instance] algebraOption
+
+@[simp]
+lemma algebraMap_option_apply (p : MvPolynomial α R) :
+    algebraMap (MvPolynomial α R) (MvPolynomial (Option α) R) p = rename some p :=
+  rfl
+
+/-- `MvPolynomial.optionEquivLeft` as an algebra homomorphism over `MvPolynomial α R`. -/
+noncomputable def optionEquivLeft' :
+    MvPolynomial (Option α) R ≃ₐ[MvPolynomial α R] (MvPolynomial α R)[X] where
+  __ := optionEquivLeft R α
+  commutes' p := by
+    induction p using MvPolynomial.induction_on with
+    | C a => simp [optionEquivLeft_C]
+    | add p q _ _ => simp_all
+    | mul_X p n hn => simp_all [optionEquivLeft_X_some]
```

Comment:
Can you provide `optionEquivLeft R α (rename some p) = C p` as its own lemma? Thanks.
and the proof is probably via `(optionEquivLeft R α).symm.injective` and `MvPolynomial.optionEquivLeft_symm_apply`.